### PR TITLE
Fix bug - was missing a call to takeMeBack in the button click

### PR DESCRIPTION
**Pull request by nichelicorn**
## Is this a fix or a feature?
First fix!

## What is the change?
Debug - button click on `Show My Poster` buttton will now switch to the homepage and display the user input values
Need an additional debut - user values do not clear when the view switches to homepage

## What does it fix?
View was not switching to home page - added call of `takeMeBack()` to the showUserPoster function
Also added a helper function to push values into the arrays - have not yet tested this functionality

## Where should the reviewer start?
Review main.js lines 195 - 209
Commented out several lines and refactored to eliminate single-use variables

## How should this be tested?
Browser - open index.html in the terminal

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ var buttonMakePoster = document.querySelector('.show-form');
 var buttonNevermind = document.querySelector('.show-main');
 var buttonBackToMain = document.querySelector('.back-to-main');
 var buttonShowPoster = document.querySelector('.make-poster');
-var imageInput = document.getElementById('poster-image-url').value;
```
Comment (nichelicorn) [inline src/main.js:22]: the .value at the end of line 22 was causing a syntaxError in the console

File: src/main.js [hunk 2]
```diff
@@ -186,17 +186,26 @@ function takeMeBack() {
   savedPostersView.classList.add('hidden');
 }
 
+function pushValues() {
+  images.push(image.src);
+  titles.push(title.innerText);
+  quotes.push(quote.innerText);
+}
+
```
Comment (nichelicorn) [inline src/main.js:194]: pushValues() has not yet been tested

File: index.html [hunk 3]
```diff
@@ -8,7 +8,7 @@
   <title>Hang In There</title>
 </head>
 <body>
-  <section class="main-poster">
+  <section class="main-poster hidden">
```
Comment (nichelicorn) [inline index.html:11]: The main poster has been set to `hidden` for debugging! This class can be removed once debugging issues are resolved

File: src/main.js (hunk 1 above)
Comment (pcmueller) [reply src/main.js:22]: Ah good catch, forgot to delete.

File: src/main.js [hunk 4]
```diff
@@ -162,13 +162,13 @@ function buildRandomPoster() {
...
   quote.innerText = randomQuote;
-    // figure out how to import/require poster.js, so that we can start using our class!
-  currentPoster = new Poster(randomImage, randomTitle, randomQuote);
+  // currentPoster = new Poster(randomImage, randomTitle, randomQuote);
 };
 
 function saveCurrentPoster(current) {
   savedPosters.push(current);
 }
+// This function may need to used savedPosters.push(currentPoster) - do we have a variable named current? I wasn't sure what this was referring to {nd}
```
Comment (pcmueller) [inline src/main.js:171]: We don't yet - I was using that as a parameter/placeholder.  I thought we might want to invoke saveCurrentPoster in another function, and so using that 'current' parameter (or whatever we choose to name it) we could pass through the Poster class iteration that we want to push into the array.

**Comment by nichelicorn** [comment]
"Need an additional *debug - user values do not clear when the view switches to homepage" --from the original PR - this may not in fact be a bug per se. I reviewed the rubric and it doesn't say anything about the form clearing so this comment can be disregarded 🌟

File: src/main.js (hunk 4 above)
Comment (nichelicorn) [reply src/main.js:171]: Nice. Thank you for clarifying! ✨

File: src/main.js [hunk 5]
```diff
@@ -186,17 +186,26 @@ function takeMeBack() {
   savedPostersView.classList.add('hidden');
 }
 
+function pushValues() {
```
Comment (pcmueller) [inline src/main.js:189]: Nice - let's test this out next.

Comment (nichelicorn) [reply src/main.js:189]: ✅ I think that would wrap up iteration 2!

File: src/main.js [hunk 6]
```diff
@@ -186,17 +186,26 @@ function takeMeBack() {
...
+  // var userTitle = titleInput.value;
+  // var userQuote = quoteInput.value;
+  // image.src = userImage;
+  // title.innerText = userTitle;
+  // quote.innerText = userQuote;
+  image.src = imageInput.value;
+  title.innerText = titleInput.value;
+  quote.innerText = quoteInput.value;
+  // console.log(userImage, userTitle, userQuote);
+  currentPoster = new Poster(imageInput.value, titleInput.value, quoteInput.value);
```
Comment (pcmueller) [inline src/main.js:207]: I see you've added takeMeBack() to currentPoster as a 'new class' instantiation parameter - will simply invoking showUserPoster() do the trick and send us back to the main page, then?

**Review by pcmueller — APPROVED** [review]
Great job on the showUserPoster() function debugging - now we should be able move on to the rest of Iteration 2.

File: src/main.js (hunk 6 above)
Comment (nichelicorn) [reply src/main.js:207]: on line 208? If this is what you're referring to, the function takeMeBack() is there to add and remove the hidden classes from those HTML sections. I'm not sure we discussed those during our call. Do you have questions about how this one works?

**Pull request merged**